src/main/java/gui/controllers/TimeLogController.java

```diff
@@ -161,19 +162,66 @@ public void setApproveButton(final Button button) {
         this.approveButton = button;
     }
 
+    /**
+     * Setter for the question asked when a chest is shown.
+     * @param label the label that needs to be set
+     */
+    public void setQuestion(final Label label) {
+        this.question = label;
+    }
+
     /**
      * Process the frames depending on the changes in cameraHandler.
      * @param now The current time.
      */
     public void processFrame(final long now) {
-        if (cameraHandler.isChestDetected()) {
-            approveButton.setVisible(true);
-            notApproveButton.setVisible(true);
-        }
         changeTime(now);
+        checkMatInformation();
         checkInformation();
     }
 
+    /**
+     * Check the information from the Mat queue for pictures of chests.
+     */
+    public void checkMatInformation() {
+        if (chestTimestamp == -1) {
+            Pair<Mat, Long> mat = informationHandler.getMatrix();
+            if (mat != null) {
+                question.setVisible(true);
+                approveButton.setVisible(true);
+                notApproveButton.setVisible(true);
+                imageView.setVisible(true);
+                BufferedImage bufferedFrame = Util.matToBufferedImage(mat.getKey());
+
+                final int newWidth = 300;
+                final int newHeight = 200;
+                BufferedImage resizedSubFrame = resizeBImage(bufferedFrame, newWidth, newHeight);
+                Image image = SwingFXUtils.toFXImage(resizedSubFrame, null);
+
+                imageView.setImage(image);
+                chestTimestamp = mat.getValue();
+            }
+        }
+    }
+
+    /**
+     * Resize the BufferedImage to fit in the screen.
+     * @param bi the BufferedImage
+     * @param newWidth the new width
+     * @param newHeight the new height
+     * @return scaledImage
+     */
+    public BufferedImage resizeBImage(final BufferedImage bi,
```

Comment:
This method could also be in util. As it does not require any specific variables.